### PR TITLE
feat: Add BrowserWindow option to hide window in Mission Control (macOS)

### PR DESCRIPTION
**Pull request by carterbs**
* There are many circumstances when app developers may want to hide their windows from mission control. E.g., full screen overlays, small helper windows, dialogs, etc.
* This PR adds the functionality, docs, and tests.

#### Description of Change
* Adds `hiddenInMissionControl` constructor option
* Adds `isHiddenInMissionControl` bw method.
* Adds `setHiddenInMissionControl` bw method.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [] `npm test` passes
- [x] tests are added (https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation is added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes:
* Adds new `setHiddenInMissionControl` API to allow developers to opt out of mission control on macOS.

### CODE REVIEW

**Comment by carterbs** [comment]
> I don't love that there are two names for this boolean, "ignoreMissionControl" and "ignoredByMissionControl". Can we settle on one or the other?

The goal was to differentiate the boolean set at construction and the value at runtime. 

"Should this window ignore mission control?"
"Is this window ignored by mission control?"

But I hear what you're saying. I'll swap it all for the past tense as I think it's cleaner, even if it's more verbose.

**Review by samuelmaddock — COMMENTED** [review]
Maybe `hiddenInMissionControl` would be more easily understood? It also closer aligns with how Apple's docs describe the behavior.

> The window floats in Spaces and hides in Mission Control.

https://developer.apple.com/documentation/appkit/nswindowcollectionbehavior/nswindowcollectionbehaviortransient

**Comment by carterbs** [comment]
Works for me. Hoping to get my dev env all set up again to make sure the
rename didn't bork something.

File: shell/browser/native_window_mac.h [hunk 1]
```diff
@@ -103,6 +103,8 @@ class NativeWindowMac : public NativeWindow,
   void SetDocumentEdited(bool edited) override;
   bool IsDocumentEdited() override;
   void SetIgnoreMouseEvents(bool ignore, bool forward) override;
+  bool IsHiddenInMissionControl() override;
+  void SetHiddenInMissionControl(bool ignore) override;
```
Comment (samuelmaddock) [inline shell/browser/native_window_mac.h:107]:
```suggestion
  void SetHiddenInMissionControl(bool hidden) override;
```

**Review by samuelmaddock — APPROVED** [review]
API LGTM

**Review by nornagon — APPROVED** [review]
API LGTM

**Review by jkleinsc — APPROVED** [review]

**Review by jkleinsc — COMMENTED** [review]
API LGTM

**Pull request merged**